api/analyticsd/service/service.go

```diff
@@ -0,0 +1,215 @@
+		}); err != nil {
+			return nil, fmt.Errorf("setting log levels: %v", err)
+		}
+	}
+
+	config := segment.Config{
+		Verbose: conf.Debug,
+	}
+	segment, err := segment.NewWithConfig(conf.SegmentAPIKey, config)
+	if err != nil {
+		return nil, fmt.Errorf("creating segment client: %v", err)
+	}
+
+	s := &Service{
+		segment: segment,
+		prefix:  conf.SegmentPrefix,
+		debug:   conf.Debug,
+	}
+
+	if conf.FilrewardsAddr != "" {
+		s.frc, err = filrewards.New(conf.FilrewardsAddr)
+		if err != nil {
+			return nil, fmt.Errorf("creating filrewards client: %v", err)
+		}
+	}
+
+	s.server = grpc.NewServer()
+	go func() {
+		pb.RegisterAnalyticsServiceServer(s.server, s)
+		if err := s.server.Serve(conf.Listener); err != nil && !errors.Is(err, grpc.ErrServerStopped) {
+			log.Errorf("serve error: %v", err)
+		}
+	}()
+
+	return s, nil
+}
+
+func (s *Service) Identify(ctx context.Context, req *pb.IdentifyRequest) (*pb.IdentifyResponse, error) {
+	if req.AccountType == 2 {
+		return &pb.IdentifyResponse{}, nil // Or should we return an error?
+	}
+	traits := segment.NewTraits()
+	traits.Set("account_type", req.AccountType)
+	traits.Set(s.prefix+"signup", "true")
+	if req.Email != "" {
+		traits.SetEmail(req.Email)
+	}
+	for key, value := range req.Properties.AsMap() {
+		traits.Set(key, value)
+	}
+	err := s.segment.Enqueue(segment.Identify{
+		UserId: req.Key,
+		Traits: traits,
+		Context: &segment.Context{
+			Extra: map[string]interface{}{
+				"active": req.Active,
+			},
+		},
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "enqueuing segment identify: %v", err)
+	}
+	return &pb.IdentifyResponse{}, nil
+}
+
+func (s *Service) Track(ctx context.Context, req *pb.TrackRequest) (*pb.TrackResponse, error) {
+	if req.AccountType == 2 {
+		return &pb.TrackResponse{}, nil // Or should we return an error?
+	}
+	props := segment.NewProperties()
+	for key, value := range req.Properties.AsMap() {
+		props.Set(key, value)
+	}
+
+	if err := s.segment.Enqueue(segment.Track{
+		UserId:     req.Key,
+		Event:      eventToString(req.Event),
+		Properties: props,
+		Context: &segment.Context{
+			Extra: map[string]interface{}{
+				"active": req.Active,
+			},
+		},
+	}); err != nil {
+		return nil, status.Errorf(codes.Internal, "enqueuing segment track: %v", err)
+	}
+
+	orgKey := ""
+	devKey := ""
+	if req.AccountType == pb.AccountType_ACCOUNT_TYPE_ORG {
+		orgKey = req.Key
+		devKey, _ = req.Properties.AsMap()["member"].(string)
+	} else if req.AccountType == pb.AccountType_ACCOUNT_TYPE_DEV {
+		devKey = req.Key
+		orgKey, _ = req.Properties.AsMap()["org_key"].(string)
+	}
+
+	// Don't process the events that are generated by s.frc.ProcessAnalyticsEvent
+	// when a reward or claim is created and sent here for analytics tracking.
+	// Also be sure that we at least have an org key since that is required.
+	if s.frc != nil && req.Event != pb.Event_EVENT_FIL_REWARD && req.Event != pb.Event_EVENT_FIL_CLAIM && orgKey != "" {
```

Comment:
Having a org key is required for calling `filrewards.ProcessAnalyticsEvent`. It's better to have dev key as well, but some events we are interested in will never have a dev key (like billing usage events), so I let that pass if dev key isn't present. The problem, which I mentioned in slack yesterday, is that js client actions that use an org API key will never have a dev key, which subverts some of the logic we'd like to apply during reward unlocking. I'll comment more about that in the fil rewards service code.